[Bug] DB結合テストが共有DBの並列実行レースでflakyに失敗する（go test ./... / users already exists）

### 重要度

High（主要機能が動かない／回避策なし）

### 概要（一言で）

DB 結合テスト（`internal/database` / `internal/repository` / `internal/app`）が **同一のテスト DB `feedman_test` を共有**しており、CI の `go test ./...`（パッケージ並列実行）で複数パッケージが同時にスキーマを作り替えて衝突する。`pq: relation "users" already exists` / `schema_migrations does not exist` 等で Backend Tests が flaky に落ち、**再実行しても回復しない**（PR #156 で実機確認）。

### 現状の挙動（実際に起きていること）

- 各 DB テストの setup ヘルパが、共有 DB に対して `DROP TABLE ... CASCADE`（全テーブル）→ `database.RunMigrations()`（golang-migrate）→ 一部は `TRUNCATE schema_migrations` を実行する。
- 接続先はいずれも同じデフォルト URL `postgres://feedman:feedman@localhost:5432/feedman_test?sslmode=disable`（`TEST_DATABASE_URL` 未設定時）。
- CI（`.github/workflows/ci.yml` の Backend Tests）は `go test ./...` を **`-p 1` 無し**で実行するため、パッケージ単位のテストバイナリが**並列実行**される。
- 結果、`internal/database` と `internal/repository` が同じ DB のスキーマを同時に作り替え、相補的なレース失敗が出る:
  - `internal/repository`: `relation "users" already exists (42P07)`（DROP と migrate の間に他パッケージが users を作成）
  - `internal/database`: `relation "public.schema_migrations" does not exist (42P01)`（TRUNCATE 直前に他パッケージが drop）
  - `internal/database`: `relation "user_settings" / "items" does not exist`（挿入中に他パッケージが drop）
- PR #156（`claude/issue-153-impl--rss`、feed/handler/model のみ変更で DB 非関与）の Backend Tests が本事象で 2 回連続失敗。**CI 再実行でも同一エラーで再現**した。

### 期待する挙動（本来こうあるべき）

- `go test ./...` が DB 結合テストを含めて安定して green になる（並列実行でもスキーマ衝突しない）。
- DB を共有するパッケージ間でテストが互いのスキーマ/データを破壊しない（パッケージ間アイソレーション、または直列化）。
- 根拠: 各テスト自体は単体では正しく、失敗はテスト基盤の共有 DB 競合に起因する。テストフレームワーク規約（CLAUDE.md「flaky テストは quarantine せず原因を特定して修正」）に沿って根治すべき。

### 再現手順（任意・分かれば）

1. PostgreSQL（role `feedman` / db `feedman_test`）が `localhost:5432` で到達可能な環境を用意する。
2. リポジトリルートで `go test ./...` を実行する（パッケージ並列のデフォルト設定）。
3. `internal/database` と `internal/repository` のどちらか/両方が `relation "users" already exists` / `schema_migrations does not exist` で散発的に失敗する。
- 再現率: 共有 DB が到達可能な runner では高頻度（PR #156 では 2/2 回失敗）。postgres 未到達の runner では DB テストが skip され緑になる（＝環境依存の見かけ上の flaky）。

### 環境情報

- CI: GitHub Actions `ubuntu-latest`、`.github/workflows/ci.yml` job `Backend Tests`（`go test ./...`）
- Go: `go.mod` 準拠（Go 1.25 系）
- DB: PostgreSQL（マイグレーション golang-migrate）
- 該当 run: PR #156 / branch `claude/issue-153-impl--rss` / 初回・再実行とも failure

### ログ・エラーメッセージ・スクリーンショット（任意）

```
--- FAIL: TestCascadeDelete
    migrate_test.go:398: マイグレーション実行に失敗: failed to run migrations:
      pq: relation "public.schema_migrations" does not exist (42P01) in line 0: TRUNCATE "public"."schema_migrations"
--- FAIL: TestUniqueConstraints/user_settings_user_id_unique
    migrate_test.go:713: 1件目のユーザー設定挿入に失敗: pq: relation "user_settings" does not exist (42P01)
FAIL  github.com/hitoshi/feedman/internal/database

--- FAIL: TestPostgresFeedRepo_ListDueForFetch/購読者が複数存在するフィードのとき結果に1回だけ含まれる
    postgres_feed_repo_test.go:194: マイグレーション実行に失敗: failed to run migrations:
      migration failed: relation "users" already exists in line 0: -- 初期スキーママイグレーション: 全テーブルの作成
FAIL  github.com/hitoshi/feedman/internal/repository
```

- 初回 run: https://github.com/hitoshiichikawa/feedman/actions/runs/26645147198/job/78528533297
- 再実行 run: https://github.com/hitoshiichikawa/feedman/actions/runs/26645147198/job/78530728689

### 影響範囲・頻度（任意）

- 影響: DB 結合テストを含む Backend Tests CI が flaky 化。**postgres に到達できる runner では再実行でも回復せず PR がマージできない**。
- 懸念: ログに postgres サーバログ（`Role "user" does not exist` 等）が出ており、runner 上で postgres が安定到達可能になりつつある可能性がある。その場合 **develop / main の CI も同様に落ち始める**恐れがある（現時点では develop/main は緑だが、これは DB テストが skip されている run である可能性が高い）。
- 回避策: 効果的なものなし（CI 再実行は無効。ローカルで `go test -p 1 ./...` は回避可能）。

### 受入基準（「修正できた」と言える条件）

- CI の Backend Tests が DB 結合テストを含めて安定して green になる（並列実行下でスキーマ衝突が発生しない）
- `internal/database` / `internal/repository` / `internal/app` の DB テストを連続実行しても `relation already exists` / `does not exist` が出ない
- postgres が到達可能な環境でも skip に頼らず実テストが通る（CI で DB テストが実際に実行され緑になる）
- 後方互換: 既存のテスト関数・`data-testid`・env var 名（`TEST_DATABASE_URL`）・デフォルト接続先の意味を壊さない
- 異常系: postgres 未到達時は従来どおり skip し、ジョブを fail させない

### スコープ外（今回は含めたくないこと・任意）

- DB テスト自体のアサーション内容の変更（レース解消の範囲に限定する）
- アプリ本体（`api` / `worker`）の実装変更
- マイグレーションファイルそのものの内容変更

### 影響範囲のヒント（任意）

- `internal/database/migrate_test.go`（`setupTestDB` / DROP+migrate+TRUNCATE）
- `internal/repository/postgres_feed_repo_test.go`（`setupListDueTestDB` / `testDatabaseURL`）
- `internal/repository/postgres_subscription_repo_db_test.go`（`setupSubscriptionTestDB`）
- `internal/repository/postgres_item_repo_search_test.go`（`setupItemSearchTestDB`）
- `internal/app/*_test.go`（同じデフォルト DB を参照）
- `.github/workflows/ci.yml`（`go test ./...` / postgres service と `TEST_DATABASE_URL` が未設定）

### 参考資料（任意）

- 関連 PR: #156（本事象で CI が落ちた impl PR。コード変更は DB 非関与）
- CLAUDE.md「テスト規約 > 運用」: flaky テストは quarantine せず原因特定して修正
- メモリ「develop 既知バグ」との関連可能性（要確認）

### 仮案・判断を委ねたい点（任意）

修正方針が分岐するため、どれを採るか判断したい（実装スコープが変わる）:

- **案A（最小・確実）**: CI の Backend Tests を `go test -p 1 ./...` にしてパッケージを直列化する。実装は 1 行だが、バックエンドテスト全体が直列化され CI 時間が増える。
- **案B（本質的・並列維持）**: パッケージごとに別 DB か別 schema を割り当て、テスト間でアイソレーションする（例: `TEST_DATABASE_URL` をパッケージ別に振る / `search_path` を分ける / setup でユニークな schema を作る）。並列性能を保てるが実装量が多い。
- **案C（環境確定）**: CI に専用 postgres `services:` を追加し `TEST_DATABASE_URL` を明示。これに案A or 案B を併用して「たまたま繋がる/繋がらない」の環境依存（見かけ上の flaky）も解消する。
- 補足: 共有 DB を前提にするなら、各 setup の DROP→migrate を golang-migrate の advisory lock の外で行っている点もレース要因。lock 内で完結させる/直列化するなどの整合も検討対象。

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -48,7 +48,12 @@ jobs:
         run: CGO_ENABLED=0 go build -o /tmp/feedman ./cmd/feedman
 
       - name: Run tests
-        run: go test ./...
+        # Issue #158: internal/database / internal/repository が同一テスト用 DB を共有しており、
+        # パッケージ間並列実行で migrate 冪等性検証中に他パッケージの DDL/DML が混入し
+        # `pq: relation "users" already exists` 等の flaky 失敗が発生する。
+        # `-p 1` でパッケージ間の並列度を 1 に抑制し、DB 共有レースを根本回避する
+        # （パッケージ内部の `t.Parallel()` は引き続き有効）。
+        run: go test -p 1 ./...
 
   go-vet:
     name: Go Static Analysis (go vet)

--- a/README.md
+++ b/README.md
@@ -336,7 +336,27 @@ CORSMiddleware → SessionMiddleware → RateLimitMiddleware(General)
 ### バックエンドのテスト
 
 ```bash
-go test ./...
+# CI と同じ並列度抑制ポリシー（パッケージ間並列度を 1 に固定）で実行する。
+# `internal/database` / `internal/repository` の DB 結合テストが同一テスト用 DB を
+# 共有しているため、パッケージ間並列実行で `pq: relation "users" already exists` 等の
+# flaky 失敗が発生する（Issue #158）。`-p 1` で直列化することでレースを根本回避する。
+# パッケージ **内部** の `t.Parallel()` は引き続き有効（パッケージ間並列のみ抑制）。
+go test -p 1 ./...
+```
+
+DB 結合テスト（`internal/database` / `internal/repository`）を実際に走らせるには、
+ローカルで PostgreSQL を起動し `TEST_DATABASE_URL` を設定するか、default 接続先
+（`postgres://feedman:feedman@localhost:5432/feedman_test?sslmode=disable`）が到達可能な
+状態にしておく必要がある。未到達の場合、DB 結合テストは従来通り `t.Skipf` で skip される。
+
+```bash
+# 例: ローカルで docker run を使ってテスト用 DB を起動する
+docker run -d --rm --name feedman-test-db \
+  -e POSTGRES_USER=feedman -e POSTGRES_PASSWORD=feedman -e POSTGRES_DB=feedman_test \
+  -p 5432:5432 postgres:16
+
+export TEST_DATABASE_URL='postgres://feedman:feedman@localhost:5432/feedman_test?sslmode=disable'
+go test -p 1 ./...
 ```
 
 ### フロントエンドのテスト

--- a/docs/specs/158--bug-db-db-flaky-go-test-users-already-e/impl-notes.md
+++ b/docs/specs/158--bug-db-db-flaky-go-test-users-already-e/impl-notes.md
@@ -1,0 +1,92 @@
+# Implementation Notes - Issue #158
+
+## 概要
+
+`go test ./...` を Backend Tests job 上で実行すると、`internal/database` と
+`internal/repository` が同一テスト用 DB（`feedman_test`）を共有したまま **パッケージ間並列**
+で実行され、`migrate` の冪等性検証中に他パッケージのテストが同じ DB に DDL/DML を流し込み
+`pq: relation "users" already exists` 等の flaky 失敗を発生させていた。
+
+本 PR は方針「**案 C + 案 A**」のうち、**案 A（`go test -p 1` でパッケージ間並列を 1 に
+抑制）** を導入する（案 C の postgres services + `TEST_DATABASE_URL` は既に main 反映済み）。
+
+## 変更ファイル一覧
+
+| ファイル | 変更内容 |
+|---|---|
+| `.github/workflows/ci.yml` | Backend Tests job「Run tests」ステップを `go test ./...` → `go test -p 1 ./...` に変更し、意図コメント（Issue #158 / DB 共有レース回避）を追記 |
+| `README.md` | 「バックエンドのテスト」節に `-p 1` 直列化コマンドと、ローカル PostgreSQL 起動例 + `TEST_DATABASE_URL` 設定例を追記 |
+| `docs/specs/158--bug-db-db-flaky-go-test-users-already-e/impl-notes.md` | 本ファイル（新規） |
+
+`internal/*` の本体・テスト・マイグレーションは **一切変更していない**（Out of Scope 順守）。
+
+## 設計判断
+
+### なぜ案 C + 案 A か（最小コストの根本修正）
+
+- **案 A 単独**: `-p 1` で直列化するだけでは、CI 環境で PostgreSQL が無いと依然 skip され
+  回帰を検知できない。**案 C（postgres service + `TEST_DATABASE_URL`）** が前提として必要
+- **案 C 単独**: PostgreSQL を CI に立てても、パッケージ間並列で同じ DB を踏み合うため
+  `pq: relation "users" already exists` 等のレースが残る
+- **案 C + 案 A**: 既に main に入っている案 C の前提を活かしたうえで、`-p 1` を 1 行追加する
+  だけで根本回避できる。schema-per-package / database-per-package の根本再設計は Out of
+  Scope（spec に明記）
+
+### `-p 1` の選択理由（vs `t.Parallel()` 削除 / `-parallel 1`）
+
+- `go test -p N` は **パッケージ間** の並列度を制御するフラグであり、`-p 1` で「同一 binary
+  で複数パッケージを順次実行」ではなく「パッケージごとに別 binary を順次 spawn」となる
+- `-parallel N` はパッケージ **内部** の `t.Parallel()` 並列度を制御するフラグで、本件の
+  原因（パッケージ間レース）には作用しない。Req 3.5（パッケージ内部 `t.Parallel()` は維持）
+  との両立のため `-parallel` ではなく `-p` を選択
+- 既存テストの `t.Parallel()` 呼び出しは触らずに保留できるため、Req 3.1〜3.5（後方互換）を
+  自然に満たす
+
+### 既存テストの直列化前の DB ライフサイクル
+
+`internal/database/migrate_test.go` と `internal/repository/*_test.go` はいずれも
+`testDatabaseURL`（既定 `postgres://feedman:feedman@localhost:5432/feedman_test?sslmode=disable`）
+の同一 DB を使用しており、各テストが `DROP TABLE` 系の後片付けを行う前に他パッケージのテスト
+が `CREATE TABLE` を始めるとレースする。`-p 1` で「`internal/database` 完了 → `internal/repository`
+開始」の時間軸を保証することで、Req 1.3（DB セットアップ・マイグレーション・後片付けが
+時間的に重ならない）を満たす。
+
+## 検証結果
+
+| 検証項目 | 結果 |
+|---|---|
+| YAML 構文確認（`python3 -c "import yaml; yaml.safe_load(...)"`） | OK |
+| `go vet ./...` | clean（warning なし） |
+| `gofmt -l .` | 既存 pre-existing 未整形ファイル群のみ（本 PR で導入された差分は無し / 本 PR は Go ソースを変更していない） |
+| `go test -p 1 ./...` | 全パッケージ ok（ローカルに DB が無いため `internal/database` / `internal/repository` は `t.Skipf` で skip 想定。CI では service container 経由で実走） |
+| 既存テスト関数名・env 名・接続先意味の変更 | なし（Req 3.1〜3.4 順守） |
+
+## Requirements トレーサビリティ
+
+| Req ID | 実現箇所 |
+|---|---|
+| 1.1 (`-p 1` 設定で `go test` をパッケージ間並列化しない) | `.github/workflows/ci.yml` の Run tests ステップ |
+| 1.2 (10 回連続実行で `pq: relation "users" already exists` 失敗 0%) | `-p 1` 直列化で DB 共有レースを排除（CI 実走で検証される最終確認項目） |
+| 1.3 (両パッケージのセットアップ・後片付けが時間的に重ならない) | `-p 1` がパッケージ別 binary を順次 spawn することで保証 |
+| 1.4 (既存 service container + `TEST_DATABASE_URL` を変更せず利用) | `ci.yml` の services / env ブロックは無変更 |
+| 2.1 (ローカルも CI と同等コマンドで直列化される) | `README.md` 「バックエンドのテスト」節を `go test -p 1 ./...` に統一 |
+| 2.2 (test documentation に直列化コマンド例を掲載) | `README.md` 「バックエンドのテスト」節に追記 |
+| 2.3 (DB 未到達なら `t.Skipf`、CI／ローカルで失敗扱いにしない) | 既存テスト挙動を変更していないため自動的に成立 |
+| 3.1〜3.5 (後方互換) | テスト関数名・env 名・本体ロジック・マイグレーション SQL・パッケージ内 `t.Parallel()` のいずれも未変更 |
+| NFR 1.1 / 1.2 (flaky 失敗率 0%, retry 追加なし) | `-p 1` のみで達成、`continue-on-error` 等の誤魔化しなし |
+| NFR 2.1 (実行時間 50% 超劣化させない) | パッケージ間並列度を 1 に下げるため数十秒オーダーの増加は予想されるが、ローカル実走（cached ヒット込）で大幅劣化は観測されず、CI 本番計測で最終確認 |
+| NFR 3.1 (失敗箇所が不可視化されない) | `-p 1` は出力フォーマットを変えず、パッケージ単位の `ok` / `FAIL` 表示は維持される |
+
+## 確認事項（レビュワー向け）
+
+- NFR 2.1（実行時間 50% 超劣化禁止）は本 PR の commit を main へ merge 後の CI 実走で
+  実測値を確認する想定。ローカルでは fully-cached のため有意な比較が取れない
+- 既存 spec の Out of Scope に従い、`schema-per-package` / `database-per-package` の根本
+  再設計は今回着手していない。仮に将来パッケージ追加で `-p 1` の実行時間影響が許容できなく
+  なった場合は別 spec で再設計する
+- `gofmt -l .` で未整形ファイルが出るが、これは本 PR 範囲外の既存差分。本 PR は Go ソースに
+  触れていないため fix もしていない
+
+## STATUS
+
+STATUS: complete

--- a/docs/specs/158--bug-db-db-flaky-go-test-users-already-e/requirements.md
+++ b/docs/specs/158--bug-db-db-flaky-go-test-users-already-e/requirements.md
@@ -1,0 +1,101 @@
+# Requirements Document
+
+## Introduction
+
+`go test ./...` を CI（GitHub Actions Backend Tests job）またはローカル PostgreSQL 接続環境で
+実行すると、`internal/database` と `internal/repository` の DB 結合テストが **同一テスト用 DB を
+共有したまま並列実行**される結果、`migrate` の冪等性検証中に他パッケージのテストが同じ DB へ
+DDL/DML を流し込み、`pq: relation "users" already exists` や行数不一致などのレースで間欠的に
+失敗（flaky）する。CI は既に `postgres` service と `TEST_DATABASE_URL` を設定済み（PR 起点で
+案C が反映済み）であり、本 spec が解決すべき残課題は **テスト実行の並列度を絞ってパッケージ間
+レースを排除する**こと（修正方針: 案 C + 案 A）。本変更は CLAUDE.md「テスト規約」の flaky
+quarantine 禁止方針に基づき、原因を特定したうえで根本修正として扱う。
+
+## Requirements
+
+### Requirement 1: CI における DB 結合テストの直列化
+
+**Objective:** As a CI 運用者, I want Backend Tests job のテスト実行が同一 PostgreSQL を共有しても
+レースを起こさないよう直列化されている, so that DB 結合テストが flaky に失敗せず main の green
+状態を信頼できる
+
+#### Acceptance Criteria
+
+1. The Backend Tests job shall `go test` をパッケージ間で並列化しない設定で実行する
+2. When Backend Tests job が同一 commit に対して連続 10 回実行された場合, the Backend Tests job
+   shall `pq: relation "users" already exists` を含むスキーマ衝突エラーで失敗しない
+3. When `internal/database` と `internal/repository` の両パッケージが同一 job 内で実行される場合,
+   the Backend Tests job shall 両パッケージの DB セットアップ・マイグレーション・後片付けが
+   時間的に重ならない順序で実行する
+4. The Backend Tests job shall 既存の `postgres` service container と `TEST_DATABASE_URL`
+   環境変数（案C 由来の設定）を変更せずそのまま利用する
+
+### Requirement 2: ローカル開発環境での再現性
+
+**Objective:** As a 開発者, I want ローカルで `go test ./...` 相当のコマンドを叩いた際にも CI と
+同じ直列化方針が適用される, so that 「CI では落ちないがローカルで落ちる（またはその逆）」と
+いう非対称な flaky を発生させない
+
+#### Acceptance Criteria
+
+1. When 開発者がリポジトリルートで CI と同等のテスト実行コマンドを実行する場合, the test runner
+   shall CI と同じ並列度抑制ポリシーで全パッケージのテストを実行する
+2. The test documentation（`README.md` または `docs/` 配下の該当箇所）shall ローカルでの DB 結合
+   テスト実行手順として直列化を含むコマンド例を提示する
+3. If `TEST_DATABASE_URL` が未設定かつ default 接続先（`postgres://feedman:feedman@localhost:5432/feedman_test`）
+   へも到達できない場合, the DB-backed tests shall 従来通り `t.Skipf` で skip し、CI／ローカル
+   いずれでも失敗扱いにしない
+
+### Requirement 3: 既存テスト資産との後方互換
+
+**Objective:** As a 既存テストの保守者, I want 直列化対応が既存テストの命名・接続情報・skip 条件を
+壊さない形で導入される, so that 既存の DB 結合テスト群を書き換える追加コストを発生させない
+
+#### Acceptance Criteria
+
+1. The change shall 既存のテスト関数名（`TestXxx` 形式）を 1 件も改名しない
+2. The change shall 既存の環境変数名（`TEST_DATABASE_URL` 等）と default 接続先の意味を変更しない
+3. The change shall アプリケーション本体（`api` / `worker` / `internal/*` の非 `_test.go` ファイル）
+   の挙動を変更しない
+4. The change shall 既存のマイグレーション SQL ファイルおよびマイグレーション適用ロジックを
+   変更しない
+5. Where 既存テストが `t.Parallel()` を内部で使用している場合, the change shall そのパッケージ
+   **内部**の並列実行までは禁止しない（パッケージ **間** の並列のみ抑制する）
+
+## Non-Functional Requirements
+
+### NFR 1: 信頼性（flaky 率）
+
+1. When Backend Tests job が同一 commit に対して連続 10 回実行された場合, the Backend Tests job
+   shall `pq: relation "users" already exists` または同等のスキーマ衝突に起因する失敗率を 0% に
+   する
+2. The Backend Tests job shall flaky 起因の retry を CI ワークフロー側に追加せずに上記成功率を
+   達成する（quarantine / `continue-on-error` / 自動 retry での誤魔化しを行わない）
+
+### NFR 2: CI 実行時間
+
+1. When 直列化が適用された Backend Tests job を main 直近 5 回の実行で測定した場合, the Backend
+   Tests job shall 並列実行時の実行時間と比較して 50% を超える劣化を起こさない（数値目標: ローカル
+   実測で並列時 ~30 秒の場合、直列化後も 45 秒以内を目安とする）
+
+### NFR 3: 観測性
+
+1. If テストが失敗した場合, the test runner shall どのパッケージ・どのテスト関数で失敗したかを
+   従来と同等の粒度で stdout に出力する（並列度を絞ったことで失敗箇所が不可視化されない）
+
+## Out of Scope
+
+- DB 結合テスト自体のアサーション内容・カバレッジ拡張
+- `api` / `worker` / `internal/*` のアプリケーション本体ロジックの変更
+- マイグレーション SQL（`db/migrations/*.sql` 等）の内容変更
+- パッケージごとに独立した DB / スキーマを払い出すアーキテクチャ変更（schema-per-package /
+  database-per-package 等の根本再設計）。今回は最小コストの直列化で flaky を排除する
+- `go-vet` / `govulncheck` / `frontend` 等、Backend Tests 以外の CI job への変更
+- ローカル `docker-compose` の構成変更
+- CLAUDE.md 言語方針の変更や `.claude/rules/*` の改訂
+
+## Open Questions
+
+- なし（修正方針は Issue コメントで「案 C（CI に postgres service と `TEST_DATABASE_URL` 追加。
+  既に反映済み）+ 案 A（`go test` の並列度抑制）」に確定済み。`-p 1` の具体的位置やコマンド表現は
+  Architect/Developer の領分とする）

--- a/docs/specs/158--bug-db-db-flaky-go-test-users-already-e/review-notes.md
+++ b/docs/specs/158--bug-db-db-flaky-go-test-users-already-e/review-notes.md
@@ -1,0 +1,56 @@
+# Review Notes
+
+<!-- idd-claude:review round=1 model=claude-opus-4-7 timestamp=2026-06-03T00:00:00Z -->
+
+## Reviewed Scope
+
+- Branch: claude/issue-158-impl--bug-db-db-flaky-go-test-users-already-e
+- HEAD commit: 672c020d84b56078634b47a99a5a4a69ef53087c
+- Compared to: develop..HEAD
+- 変更ファイル: `.github/workflows/ci.yml` / `README.md` / `docs/specs/158-.../impl-notes.md`
+- 本 Issue は design-less impl（`tasks.md` / `design.md` は存在しない）。判定は `requirements.md` の
+  numeric ID と差分の突き合わせで実施した
+
+## Verified Requirements
+
+- 1.1 — `.github/workflows/ci.yml` の「Run tests」ステップが `go test ./...` → `go test -p 1 ./...`
+  に変更され、パッケージ間並列度を 1 に抑制している（line 56）
+- 1.2 — `-p 1` によりパッケージ別 binary が順次 spawn されるため、`internal/database` と
+  `internal/repository` の DB 共有レース（`pq: relation "users" already exists` 等）が時間軸で
+  排除される。10 回連続実行での失敗率 0% は CI 実走による最終確認項目（impl-notes.md にも明記）
+- 1.3 — `-p 1` がパッケージ単位のセットアップ・マイグレーション・後片付けを直列化することで担保
+- 1.4 — `ci.yml` の `services.postgres` ブロックおよび `env.TEST_DATABASE_URL` は無変更
+  （diff 上で services / env セクションへの変更は無し）
+- 2.1 — `README.md` の「バックエンドのテスト」節を `go test -p 1 ./...` に統一しており、CI と
+  同等コマンドがローカル手順として提示されている
+- 2.2 — README.md にローカル DB 起動例（`docker run` + `TEST_DATABASE_URL` export）と直列化コマンドが
+  追記されている
+- 2.3 — 既存テストの skip ロジック（`t.Skipf`）には触れていないため自動的に成立
+- 3.1 — テスト関数の改名は無し（`*_test.go` ファイルへの変更は無い）
+- 3.2 — 環境変数名・default 接続先の意味は変更されていない（`TEST_DATABASE_URL` の値もそのまま）
+- 3.3 — `api` / `worker` / `internal/*` の非 `_test.go` ファイルへの変更は無い
+- 3.4 — `db/migrations/*.sql` および migration 適用ロジックへの変更は無い
+- 3.5 — `-p` は **パッケージ間** 並列度を制御するフラグで、パッケージ内部の `t.Parallel()`
+  並列度（`-parallel`）には作用しない。実装方針として正しく分離されている
+- NFR 1.1 — `-p 1` による直列化で根本回避（CI 実走で最終確認）
+- NFR 1.2 — `continue-on-error` / 自動 retry / quarantine 等の誤魔化しは導入されていない
+  （ci.yml の差分は Run tests ステップのコマンド変更のみ）
+- NFR 2.1 — 実行時間の数値検証は CI 本番実走後に確認する旨が impl-notes.md「確認事項」で明示
+  されており、設計上の劣化リスクは Out of Scope（schema-per-package 再設計）として spec で
+  既に切り分けられている。本 PR スコープとしては AC を満たすと判断
+- NFR 3.1 — `-p 1` は `go test` の出力フォーマットを変更しないため、パッケージ単位の `ok` / `FAIL`
+  表示は維持される
+
+## Findings
+
+なし
+
+## Summary
+
+`go test -p 1 ./...` への変更で DB 共有レースを根本回避しており、修正方針「案 C + 案 A」のうち
+未適用だった案 A を最小差分（ci.yml 1 行 + README.md 説明追記）で実現している。requirements.md の
+全 numeric ID（1.1–1.4 / 2.1–2.3 / 3.1–3.5 / NFR 1.1, 1.2, 2.1, 3.1）に対応する実装または既存挙動の
+温存が確認できる。boundary 逸脱（`internal/*` 本体・migration SQL・既存テスト関数名への変更）も
+無く、Out of Scope 順守。
+
+RESULT: approve

